### PR TITLE
Add .gt() and .lt()

### PR DESCRIPTION
**Pull request by wataori**
#129

We've wanted to use .gt() and .lt() like jQuery and I added these features.

I use the ` Array.slice() ` function to add these .gt() and .lt() functions.
these functions may be different to the concept of the cheerio if it is ` O(n) `, but I don't know :-(
Please tell us if someone know about this.

### CODE REVIEW

File: lib/api/traversing.js [hunk 1]
```diff
@@ -353,6 +353,14 @@ exports.eq = function(i) {
   return this[i] ? this._make(this[i]) : this._make([]);
 };
 
+exports.gt = function(i) {
+  return this.length > 1 ? this._make(this.slice(i + 1)) : this;
```
Comment (fb55) [inline lib/api/traversing.js]: I'm not sure about the behavior for `length <= 1`. Shouldn't an empty collection be returned?

**Comment by jugglinmike** [comment]
These methods are not implemented in jQuery, so including them here would only allow people to write less portable code. I know you are making up for the lack of the associated pseudo selectors (an existing portability concern), but the best response to that would be to implement those in the CSSselect library.

That may be a little more challenging, but the good news is that we now have [explicit instructions for extending Cheerio](https://github.com/cheeriojs/cheerio#plugins). You can follow those to implement any custom API you would like for your application, including `lt` and `gt`.

File: lib/api/traversing.js (hunk 1 above)
Comment (wataori) [inline lib/api/traversing.js]: Uhhh...Exactly. Thanks.

**Comment by wataori** [comment]
> These methods are not implemented in jQuery, 

Do you mean that these are implemented in Sizzle? I'm a little confused. But OK. I'd like to define and try these methods at prototype object in my own apps. I'll push commits if these get succesful.

**Comment by jugglinmike** [comment]
@wataori Correct. For instance, there is [a `:gt` selector](http://api.jquery.com/gt-selector/), but there is no `gt` method. As noted in the relevant jQuery documentation pages, if you'd like to use a method, `slice` offers equivalent behavior. And good news: [Cheerio implements the `slice` method](https://github.com/cheeriojs/cheerio#slice-start-end-) :)

**Comment by wataori** [comment]
@jugglinmike 
>  Cheerio implements the slice method :)

Great!!

**Pull request closed without merging**